modules/file-transfer/src/main/java/org/apache/ignite/internal/network/file/FileTransferServiceImpl.java

```diff
@@ -242,87 +249,104 @@ private void processUploadRequest(FileUploadRequest message, String senderConsis
+                }, executorService);
     }
 
     private void processDownloadRequest(FileDownloadRequest message, String senderConsistentId, Long correlationId) {
         supplyAsync(() -> getFileProvider(message.identifier()), executorService)
                 .thenCompose(provider -> provider.files(message.identifier()))
                 .whenComplete((files, e) -> {
                     if (e != null) {
-                        LOG.error("Failed to get files for download. Metadata: {}", message.identifier(), e);
+                        LOG.error("Failed to get files for download. Metadata: {}", e, message.identifier());
 
                         FileDownloadResponse response = messageFactory.fileDownloadResponse()
-                                .error(buildError(e))
+                                .error(fromThrowable(messageFactory, e))
                                 .build();
 
                         messagingService.respond(senderConsistentId, FILE_TRANSFER_CHANNEL, response, correlationId);
                     } else if (files.isEmpty()) {
                         LOG.warn("No files to download. Metadata: {}", message.identifier());
 
                         FileDownloadResponse response = messageFactory.fileDownloadResponse()
-                                .error(buildError(new FileTransferException("No files to download")))
+                                .error(fromThrowable(messageFactory, new FileTransferException("No files to download")))
                                 .build();
 
                         messagingService.respond(senderConsistentId, FILE_TRANSFER_CHANNEL, response, correlationId);
                     } else {
                         FileDownloadResponse response = messageFactory.fileDownloadResponse()
-                                .headers(fromPaths(messageFactory, files))
                                 .build();
 
                         messagingService.respond(senderConsistentId, FILE_TRANSFER_CHANNEL, response, correlationId)
-                                .thenComposeAsync(v -> sendFiles(senderConsistentId, message.transferId(), files), executorService);
+                                .thenComposeAsync(v -> {
+                                    return transferFilesToNode(senderConsistentId, message.transferId(), message.identifier(), files);
+                                }, executorService);
                     }
                 });
     }
 
-    private void processFileChunkMessage(FileChunkMessage message) {
-        runAsync(() -> fileReceiver.receiveFileChunk(message), executorService);
+    private void processFileChunkMessage(FileChunkMessage message, String senderConsistentId, long correlationId) {
+        runAsync(() -> fileReceiver.receiveFileChunk(message), executorService)
+                .whenComplete((v, e) -> {
+                    if (e != null) {
+                        LOG.error("Failed to receive file chunk. Transfer ID: {}", e, message.transferId());
```

Comment:
What does it mean, "failed to receive"? You have received it, but wasn't able to process the data. It's not the same

Reply:
By the way, I don't think that this is how we should format our log messages, usually we have all the extra data in brackets, not as a separate sentence. This comment applied to a lot of your log messages